lib/matplotlib/testing/conftest.py

```diff
@@ -125,3 +125,39 @@ def test_imshow_xarray(xr):
 
     xr = pytest.importorskip('xarray')
     return xr
+
+
+def _text_placeholders(monkeypatch):
+    from matplotlib.patches import Rectangle
+
+    def patched_get_text_metrics_with_cache(renderer, text, fontprop, ismath, dpi):
+        """
+        Replace ``_get_text_metrics_with_cache`` with fixed results.
+
+        The usual ``renderer.get_text_width_height_descent`` would depend on font
+        metrics; instead the fixed results are based on font size and the length of the
+        string only.
+        """
+        height = fontprop.get_size()
+        width = len(text) * height / 1.618  # Golden ratio for character size.
+        return width, height, 0
```

Comment:
Texts seem to generally, a somewhat low x-height:

![grafik](https://github.com/user-attachments/assets/a93a5345-623b-47e7-bfc2-5f4c63bfaf0f)

Could this be related to descent=0?

A simple and rougly correct assumtion would be `descent = height // 4`.

Reply:
It's not really the x-height but the entire height of the text that's smaller. Technically, I believe `get_window_extent` is in pixels, and font size is in points, so I could run this through `renderer.points_to_pixels`, but I found that to be much too large (because this calculation assumes a monospace font while we aren't using one.)

I think `descent=0` is still okay though, as the legend handle is aligned to the centre of the text and the placeholder box. If I set `descent` to anything else, then the line is not longer aligned as such.